### ISSUE TITLE
Better formatted error messages in service responses

### ISSUE THREAD

**Issue by venki09** [cerner/beadledom#24]
Our services currently return responses with 40x-50x if we encounter any client-side/server-side errors. The response now contains a plain text message explaining what's bad in the request and wraps it in to a Jax-rs response object. For example: all 400 responses use the [JaxRsParamCondition](https://github.com/cerner/beadledom/blob/2.4/jaxrs/src/main/java/com/cerner/beadledom/jaxrs/JaxRsParamConditions.java#L44-L50) method provided by beadledom. 

It would be helpful to have a more structured error message [Google API as an example]( https://developers.google.com/drive/v2/web/handle-errors) passed to the service consumer which they can parse and take necessary actions.

**Comment by b-boogaard** [comment]
@venki09 is the idea that beadledom provides a serializable model that supports this format (i.e. a jackson model)?

**Comment by venki09** [comment]
@b-boogaard : Yeah that is the idea. Let me know if there are better ways of handling this. Thanks

**Comment by b-boogaard** [comment]
@venki09 I am pretty sure there are plans for something like this. The blocker is defining and settling on what the structure should actually be.

**Comment by Nino55N** [comment]
We actually ran into this being a problem with the last few releases of hi-service_client: https://jira2.cerner.com/browse/HEALTHEINT-23727. I have proposed that Infra make a small update to hi-service_client in order to get us around this issue for the time being but a services update to return JSON in these instances would be the best solution.

**Comment by L2lucidream** [comment]
+1 to structure google is using, or even a simplified version of it.
```java
{
  "error": {
    "code": 500,
    "message": "my error message" 
  }
}
```

**Comment by johnlcox** [comment]
What I like about Google's error structure is that the `errors` array could be considered optional, so simple error messages could use the structure you have shown above, but something that is performing full validation on inputs may return back a full list of everything that needs to be correct in order to send a valid request.

**Comment by bjtieman** [comment]
Including a correlation id to link the error reported to the client to the error logged by the server would be helpful.  I'm otherwise in favor of the Google API.

**Comment by johnlcox** [comment]
The client should be setting the correlation id header when making the request, so that information should already be available to the client.

**Comment by johnlcox** [comment]
Fixed by 3ceffecc1f67cb5c4fa7a4cae51174571e6717ac
